var/spack/repos/builtin/packages/fakexrandr/package.py

```diff
@@ -44,3 +44,10 @@ def edit(self, spec, prefix):
 
         if 'platform=darwin' in spec:
             makefile.filter('ldconfig', '')
+
+    # In Makefile, install commands check the target dir.
+    # If it does not exist, process will stop.
+    @run_before('install')
+    def make_target_dir(self):
+        mkdir(self.prefix.lib)
+        mkdir(self.prefix.bin)
```

Comment:
Do you mean:
```suggestion
        mkdirp(self.prefix.lib)
        mkdirp(self.prefix.bin)
```
? If you meant `mkdir` I think we should check why it is working for you. Since we don't define that function in Spack it is probably leaking from a dependency.

Reply:
yes, you are right. I have look the spack src, it is better to use 'mkdirp'.  But the 'mkdir ' also looks good,  you could find so many uses in 'package.py' files. I will change it. Thank you.